pyscriptjs/src/main.ts

```diff
@@ -248,65 +249,112 @@ export class PyScriptApp {
-        logger.info('Python plugins to fetch: ', plugins);
+        logger.info('Plugins to fetch: ', plugins);
         for (const singleFile of plugins) {
             logger.info(`  fetching plugins: ${singleFile}`);
-            try {
-                const pathArr = singleFile.split('/');
-                const filename = pathArr.pop();
-                // TODO: Would be probably be better to store plugins somewhere like /plugins/python/ or similar
-                const destPath = `./${filename}`;
-                await runtime.loadFromFile(destPath, singleFile);
-
-                //refresh module cache before trying to import module files into runtime
-                runtime.invalidate_module_path_cache()
-
-                const modulename = singleFile.replace(/^.*[\\/]/, '').replace('.py', '');
-
-                console.log(`importing ${modulename}`);
-                // TODO: This is very specific to Pyodide API and will not work for other interpreters,
-                //       when we add support for other interpreters we will need to move this to the
-                //       runtime (interpreter) API level and allow each one to implement it in its own way
-                const module = runtime.interpreter.pyimport(modulename);
-                if (typeof module.plugin !== 'undefined') {
-                    const py_plugin = module.plugin;
-                    py_plugin.init(this);
-                    this.plugins.addPythonPlugin(py_plugin);
-                } else {
-                    logger.error(`Cannot find plugin on Python module ${modulename}! Python plugins \
-modules must contain a "plugin" attribute. For more information check the plugins documentation.`);
-                }
-            } catch (e) {
-                //Should we still export full error contents to console?
-                handleFetchError(<Error>e, singleFile);
+            if (singleFile.endsWith('.py')) {
+                await this.fetchPythonPlugin(runtime, singleFile);
+            } else if (singleFile.endsWith('.js')) {
+                await this.fetchJSPlugin(singleFile);
+            } else {
+                throw new UserError(
+                    ErrorCode.BAD_PLUGIN_FILE_EXTENSION,
+                    `Unable to load plugin from '${singleFile}'. ` +
+                    `Plugins need to contain a file extension and be ` +
+                    `either a python or javascript file.`);
             }
+            logger.info("All plugins fetched");
+        }
+    }
+
+    /**
+     * Fetch a javascript plugin from a filePath, it gets a blob from the
+     * fetch and creates a file from it, then we create a URL from the file
+     * so we can import it as a module.
+     *
+     * This allow us to instantiate the imported plugin with the default
+     * export in the module (the plugin class) and add it to the plugins
+     * list with `new importedPlugin()`.
+     *
+     * @param filePath - path to the js file to fetch
```

Comment:
Minor - maybe "URL of the js file to fetch", since we sometimes use "path" to refer to files in the JS/Emscripten local filesystem.